minor: Change encoding to UTF-8 in `open()` calls.

ganpa3 opened the pull request:
This allows zulint to run on Windows.

andersk commented on example-lint:
```diff
@@ -27,7 +28,8 @@ def run() -> None:
     by_lang = linter_config.list_files(file_types, exclude=EXCLUDED_FILES)
 
-    linter_config.external_linter('mypy', ['tools/run-mypy'], ['py'], pass_targets=False,
+    mypy_command = ['python', 'tools\\run-mypy'] if sys.platform == 'win32' else ['tools/run-mypy']
```
Shouldn’t `[sys.executable, 'tools/run-mypy']` do the right thing on both platforms?

This should either be a separate commit or the commit message generalized.

ganpa3 committed:
minor: Change mypy lint command.

The command earlier raised a
"WindowsError. The system cannot find specified file." error on Windows.

This commit fixes that and makes it to run on Windows.

ganpa3 committed:
minor: Change encoding to UTF-8 in `open()` calls.

open() calls were failing on Windows raising a UnicodeDecodeError error
since some files contained Unicode characters.

This commit fixes that and makes zulint to run on Windows.

timabbott commented:
This looks great implementation-wise, but @ganpa3 can you expand the commit messages to provide more of the context?  "This makes it work on Windows" is helpful, but it should explain what the actual issue was on Windows with the previous version.

ganpa3 commented:
@timabbott Updated the commit message.

timabbott commented:
Merged, thanks @ganpa3!
